src/purchase_listinator/endpoints/http/purchase_list.clj

```diff
@@ -17,22 +18,23 @@
                                :body   [out.purchases-lists/PurchaseList]}
   [{{:keys [datomic]} :component
     user-id           :user-id}]
-  (branch (misc.either/try-right
-            (let [user-id (adapters.misc/string->uuid user-id)]
-              (flows.purchase-list/get-lists user-id datomic)))
-          misc.http/->Error
-          misc.http/->Success))
+  (misc.http/default-branch
+    (misc.either/try-right
+      (-> (adapters.misc/string->uuid user-id)
+          (flows.purchase-list/get-lists datomic)
+          (->> (map adapters.out.purchase-list/internal->wire))))))
 
 (s/defn post-purchase-lists :- {:status s/Int
                                 :body   wires.purchase-list.out.purchase-list/PurchaseList}
   [{{datomic :datomic} :component
     {:keys [name]}     :json-params
     user-id            :user-id}]
-  (branch (misc.either/try-right
-            (let [user-id (adapters.misc/string->uuid user-id)]
-              (flows.purchase-list/create name user-id datomic)))
-          misc.http/->Error
-          misc.http/->Success))
+  (misc.http/default-branch
+    (misc.either/try-right
+      (-> (adapters.misc/string->uuid user-id)
+          (flows.purchase-list/create name datomic)
+          :right
```

Comment:
Instead of calling `:right`, you can use the `misc.http/default-branch-adapter`, giving the adapter as the second parameter